### PR TITLE
Credential: conditions + cluster identity watcher

### LINKED ISSUE #393: Add conditions to Credential CR's status
Add `[]metav1.Condition` field and corresponding logic in the Credential CR

### LINKED ISSUE #529: Credential: track changes in referenced object
Currently the object referenced in `IdentityRef` is not tracked by the credential controller, i.e. when it changes there is no immediate update of Credential object.

This should be fixed and Credential object must be reconciled when object referenced in `IdentityRef` gets updated. 

This is also should enable credential propagation only if they were updated and not on every reconcile of `ManagedCluster`.

### PR DESCRIPTION
**Pull request by a13x5**
Resolves #393 
Resolves #529

### CODE REVIEW

**Comment by a13x5** [comment]
Initial idea with Watcher is scrapped, simple requeue is used instead.
`SyncPeriod` is not used, since it's a cache parameter and will be applied to all controllers of the manager (which is suboptimal)

Idea to run credential propagation only in case of update is also scrapped, since currently there is no way to determine for sure if secrets were actually updated (because of how cluster identities are implemented).

File: internal/controller/credential_controller.go [hunk 1]
```diff
@@ -51,33 +61,43 @@ func (r *CredentialReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		Name:      cred.Spec.IdentityRef.Name,
 		Namespace: cred.Spec.IdentityRef.Namespace,
 	}, clIdty); err != nil {
+		errMsg := fmt.Sprintf("Failed to get ClusterIdentity %s: %s", cred.Spec.IdentityRef.Name, err)
```
[the remark this answers is not in the document]
Comment (a13x5) [reply internal/controller/credential_controller.go:64]: The only cluster identity which has namespace is the azure one and in the docs we hinting to create it in `hmc-system`. 
Based on that I decided to skip namespace altogether

**Comment by a13x5** [comment]
> This might have been discussed before, but I’m curious to understand it better. In https://github.com/Mirantis/hmc/issues/529, it was mentioned that This should enable credential propagation only if they were updated, and not on every reconcile of ManagedCluster. Was it intended to be implemented, or considered unnecessary?

Initially I though that it would be great to skip propagation reconcile if there were no changes in the credentials. The problem is we can't track them correctly: the secret change doesn't get reflected in the cluster identity and thus credential will not be updated. More so - some cluster identities don't even have a status.

> Additionally, could we consider triggering a ManagedCluster reconciliation to update the CredentialReady condition when the referenced Credential status is updated?

Yep that's a good catch, looks like I forgot to add credential watcher to managed cluster controller

**Review by eromanova — APPROVED** [review]
Got it. Thanks for the explanation, Alexey. LGTM, but I'd ask Andrey for a final review.

**Review by Kshatrix — APPROVED** [review]

**Pull request merged**